### PR TITLE
Rename generic run-on flag into platform

### LINKED ISSUE #6318: Rename generic `run-on` flag into `platform`
As an odo user, I want the experimental generic `--run-on` flag to be renamed into `--platform` because it makes more sense for listing, deleting, or describing components.

## Acceptance Criteria
- [ ] `--run-on` should be renamed into `--platform` everywhere (help output, ...)
- [ ] Internal code (e.g., variables names) should reflect that change as well
- [ ] `--platform` should be experimental
- [ ] `--platform` should be documented in the [list of experimental features](https://odo.dev/docs/user-guides/advanced/experimental-mode#list-of-experimental-features)

## Related Epic
https://github.com/redhat-developer/odo/issues/5871

Needed for the following issues:
- https://github.com/redhat-developer/odo/issues/6296
- https://github.com/redhat-developer/odo/issues/6297
- https://github.com/redhat-developer/odo/issues/6298

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

- rename `--run-on` flag with `--platform`
- change "RUNNING ON" header with "PLATFORM" in output of `odo list component`
- duplicate `runningOn` with `platform` field in JSON output of `odo list component` and deprecate `runningOn`

**Which issue(s) this PR fixes:**

Fixes #6318 

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [x] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by feloy** [comment]
@rm3l I made some changes related to https://github.com/redhat-developer/odo/pull/6438#discussion_r1059906916
But I cannot make golangci-lint ignore the deprecated field, even with special ignore rule in 
 config file, see https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6445-Validate-tests-3272.html

Do you have any experience on this problem?

**Comment by rm3l** [comment]
> @rm3l I made some changes related to [#6438 (comment)](https://github.com/redhat-developer/odo/pull/6438#discussion_r1059906916) But I cannot make golangci-lint ignore the deprecated field, even with special ignore rule in config file, see [s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6445-Validate-tests-3272.html](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6445-Validate-tests-3272.html)
> 
> Do you have any experience on this problem?

Whoops - I missed your comment.. Yes, I remember I read a while back somewhere in golangci-lint docs that this `text` field is actually interpreted verbatim as a regexp string. I can't find the reference though..
Can you try escaping the square brackets in `allComponents[i]`?

File: .golangci.yaml [hunk 1]
```diff
@@ -83,3 +83,8 @@ issues:
       # Workaround to exclude some 'staticcheck' messages, because line-based directive does not seem to work with golangci-lint
       # See https://github.com/golangci/golangci-lint/issues/741#issuecomment-1017014331
       text: "SA1019: segment.DisableTelemetryEnv is deprecated"
+    - linters:
+        - staticcheck
+      # Workaround to exclude some 'staticcheck' messages, because line-based directive does not seem to work with golangci-lint
+      # See https://github.com/golangci/golangci-lint/issues/741#issuecomment-1017014331
+      text: 'SA1019: allComponents\[i\].RunningOn is deprecated'
```
Comment (valaparthvi) [inline .golangci.yaml:90]: :100:

**Review by rm3l — CHANGES_REQUESTED** [review]
Looks like the `--run-on` flag is still being used in some tests:

```
$ git grep 'run-on' -- ':!docs/website/blog' ':!vendor'

tests/integration/cmd_describe_component_test.go:                       res := helper.Cmd("odo", "describe", "component", "--name", "unknown-name", "--run-on", "podman", "-o", "json").
tests/integration/cmd_describe_component_test.go:                       stderr := helper.Cmd("odo", "describe", "component", "--name", "unknown-name", "--run-on", "podman").
tests/integration/cmd_devfile_list_test.go:                     It("should display component depending on experimental mode and run-on flag", func() {
tests/integration/cmd_devfile_list_test.go:                                     By("returning component not in dev mode when experimental mode is enabled with json output and run-on is cluster", func() {
```

**Review by rm3l — APPROVED** [review]

**Comment by rm3l** [comment]
```
    STEP: JSON output 01/03/23 15:03:03.978
    Running odo with args [odo describe binding -o json] and odo env: []
    [odo] E0103 15:03:04.949885   18827 memcache.go:206] couldn't get resource list for packages.operators.coreos.com/v1: forbidden: User "system:anonymous" cannot get path "/apis/packages.operators.coreos.com/v1"
    [odo] [
    [odo] 	{
    [odo] 		"name": "my-nodejs-app-cluster-sample",
    [odo] 		"spec": {
    [odo] 			"application": {
    [odo] 				"kind": "Deployment",
    [odo] 				"name": "my-nodejs-app-app",
    [odo] 				"apiVersion": "apps/v1"
    [odo] 			},
    [odo] 			"services": [
    [odo] 				{
    [odo] 					"kind": "Cluster",
    [odo] 					"namespace": "cmd-describe-list-binding-test544qzb",
    [odo] 					"name": "cluster-sample",
    [odo] 					"apiVersion": "postgresql.k8s.enterprisedb.io/v1"
    [odo] 				}
    [odo] 			],
    [odo] 			"detectBindingResources": true,
    [odo] 			"bindAsFiles": false
    [odo] 		},
    [odo] 		"status": {
    [odo] 			"bindingEnvVars": [
    [odo] 				"CLUSTER_TYPE",
    [odo] 				"CLUSTER_CA.CRT",
    [odo] 				"CLUSTER_HOST",
    [odo] 				"CLUSTER_PASSWORD",
    [odo] 				"CLUSTER_TLS.KEY",
    [odo] 				"CLUSTER_PROVIDER",
    [odo] 				"CLUSTER_TLS.CRT",
    [odo] 				"CLUSTER_USERNAME",
    [odo] 				"CLUSTER_CA.KEY",
    [odo] 				"CLUSTER_CLUSTERIP",
    [odo] 				"CLUSTER_DATABASE",
    [odo] 				"CLUSTER_PGPASS"
    [odo] 			]
    [odo] 		}
    [odo] 	}
    [odo] ]
    [odo] read: read /dev/stdin: input/output error
    Deleting project: cmd-describe-list-binding-test544qzb
    Running oc with args [oc delete project cmd-describe-list-binding-test544qzb --wait=false] and odo env: []
    [oc] project.project.openshift.io "cmd-describe-list-binding-test544qzb" deleted
    Deleting project: cmd-describe-list-binding-test544tuv
    Running oc with args [oc delete project cmd-describe-list-binding-test544tuv --wait=false] and odo env: []
    [oc] project.project.openshift.io "cmd-describe-list-binding-test544tuv" deleted
    Setting current dir to: /go/odo_1/tests/integration
    Deleting dir: /tmp/3603466537
    Deleting dir: /tmp/2955535718
  << End Captured GinkgoWriter Output

  Expected
      <string>: E0103 15:03:04.949885   18827 memcache.go:206] couldn't get resource list for packages.operators.coreos.com/v1: forbidden: User "system:anonymous" cannot get path "/apis/packages.operators.coreos.com/v1"
      
  to be empty
  In [It] at: /go/odo_1/tests/integration/cmd_describe_list_binding_test.go:252

...
Summarizing 1 Failure:
  [FAIL] odo describe/list binding command tests when creating a component with a binding as environment variables (service in namespace "binding-lpv") when Starting a Pg service when running dev session [It] should describe the binding
  /go/odo_1/tests/integration/cmd_describe_list_binding_test.go:252

Ran 343 of 603 Specs in 1094.319 seconds
FAIL! -- 342 Passed | 1 Failed | 0 Pending | 260 Skipped
```

This test is explicitly expecting the stderr output to be empty, but there was a warning on stderr. I don't think we need to check stderr here.
Overriding this since it is unrelated to this PR, but I'm keeping an eye on this test and we might need to consider handling this in a separate issue/PR.

/override OpenShift-Integration-tests/OpenShift-Integration-tests

**Comment by feloy** [comment]
/override ci/prow/v4.11-integration-e2e
Flaky e2e test

**Pull request merged**